ajax.js

```diff
@@ -25,51 +30,103 @@ define('ajax', ['qwest'], function(qwest) {
 		};
 	}
 
+	function urlEncode(data) {
+		if (data) {
+			var encodedParams = [];
+			for (var k in data) {
+				encodedParams.push(encodeURIComponent(k) + '=' + encodeURIComponent(data[k]));
+			}
+			return encodedParams.join('&');
+		}
+		return null;
+	}
+
+	function getRequestObj() {
+		if (window.XMLHttpRequest) {
+			return new XMLHttpRequest();
+		} else if (window.ActiveXObject) {
+			return new ActiveXObject("Microsoft.XMLHTTP");
+		}
+	}
+
+	function isSuccess(status) {
+		return status >= 200 && status < 300;
+	}
+
+	function handleResponse(requestObj, callbacks) {
+		if (requestObj.readyState !== 4) {
+			return;
+		}
+
+		var responseType = requestObj.getResponseHeader("Content-Type") || "";
+		var responseContent = "";
+
+		if (responseType.indexOf("json") !== -1) {
+			responseContent = JSON.parse(requestObj.responseText);
+		} else if (responseType.indexOf("xml") !== -1) {
+			responseContent = (new DOMParser()).parseFromString(requestObj.responseText,'text/xml');
+		} else {
+			responseContent = requestObj.responseText;
+		}
+
+		if (isSuccess(requestObj.status) && callbacks.success && typeof callbacks.success === "function") {
+			callbacks.success(responseContent, requestObj);
+		} else if (!isSuccess(requestObj.status) && callbacks.error && typeof callbacks.error === "function") {
+			callbacks.error(requestObj.statusText, requestObj);
+		}
+
+		if (callbacks.complete && typeof callbacks.complete === "function") {
+			callbacks.complete(requestObj);
+		}
+	}
+
+	function setHeaders(url, requestObj, method, configHeaders) {
+		var headerValues = {};
+		var host = url.match(/\/\/(.+?)\//);
+		var crossOrigin = host && (host[1] ? host[1] != window.location.host : false);
+
+		headerValues["Accept"] = configHeaders["Accept"] || "*/*";
+
+		if (!crossOrigin) {
+			headerValues['X-Requested-With'] = 'XMLHttpRequest';
+		}
+
+		if (method === POST) {
+			headerValues["Content-Type"] = "application/x-www-form-urlencoded; charset=UTF-8";
```

Comment:
I think this value should be a default config of this header, but I should be able to overwrite this header for another.

@erikatakahara @fellyph